src/dmd/toobj.d

```diff
@@ -358,10 +358,14 @@ void toObjFile(Dsymbol ds, bool multiobj)
                 return;
             }
 
+            const bool gentypeinfo = global.params.useTypeInfo && Type.dtypeinfo;
```

Comment:
If you merge https://github.com/dlang/druntime/pull/2184, then you'll only need to check `Type.dtypeinfo` because the runtime will no longer import the `TypeInfo` classes when compiling with -betterC.  There are other places in the DMD source code that can be simplified due to that PR.

Reply:
I'm not so sure about that. 2184 seems to be more concerned with the nothrow issue, which is unrelated to this PR's fix. It isn't ready yet, either, and this PR fix works and is pretty simple.

Reply:
2184 is actually more concerned with preventing the import of code which is not relevant to -betterC build.  The `nothrow` addition was added because -betterC code should be `nothrow` at all times.  If you want me to separate the two into different PRs, I can.  Just let me know.  And, it is all green and ready to go.

Reply:
I'm not too happy about the nothrow business, as I mentioned. But we should discuss that elsewhere, not here. I also do not believe that 2184 will resolve this issue, as note it took two different tests to get it through the test suite, not just checking typeinfo.